test/modules/test_modules_reddit.py

```diff
@@ -1,76 +1,79 @@
-
-
-@pytest.mark.parametrize('link', NON_MATCHING_URLS)
-def test_url_non_matching(link, bot):
-    line = PreTrigger(bot.nick, ':User!user@irc.libera.chat PRIVMSG #channel {}'.format(link))
-    matches = bot.rules.get_triggered_rules(bot, line)
-
-    assert len([match for match in matches if match[0].get_plugin_name() == 'reddit']) == 0
+"""Tests for Sopel's ``reddit`` plugin"""
+from __future__ import generator_stop
+
+import pytest
+
+from sopel.trigger import PreTrigger
+
+
+TMP_CONFIG = """
+[core]
+owner = Admin
+nick = Sopel
+enable =
+    reddit
+host = irc.libera.chat
+"""
+
+
+@pytest.fixture
+def bot(botfactory, configfactory):
+    settings = configfactory('default.ini', TMP_CONFIG)
+    return botfactory.preloaded(settings, ['reddit'])
+
+
+MATCHING_URLS = (
+    # URLs the reddit plugin is expected to handle
+    # Should match ONCE each, no more, no less
+    'https://redd.it/123456',
+    'https://redd.it/123456/',
+    'https://reddit.com/123456',
+    'https://reddit.com/123456/',
+    'https://reddit.com/r/subname',
+    'https://reddit.com/r/subname/',
+    'https://www.reddit.com/r/subname',
+    'https://www.reddit.com/r/subname/',
+    'https://reddit.com/comments/123456',
+    'https://reddit.com/comments/123456/',
+    'https://www.reddit.com/comments/123456',
+    'https://www.reddit.com/comments/123456/',
+    'https://reddit.com/r/subname/comments/123456',
+    'https://reddit.com/r/subname/comments/123456/',
+    'https://www.reddit.com/comments/123456?param=value',
+    'https://www.reddit.com/comments/123456/?param=value',
+    'https://reddit.com/r/subname/comments/123456?param=value',
+    'https://reddit.com/r/subname/comments/123456/?param=value',
+    'https://www.reddit.com/r/subname/comments/123456',
+    'https://www.reddit.com/r/subname/comments/123456/',
+    'https://reddit.com/r/subname/comments/123456/post_title_slug/234567',
+    'https://reddit.com/r/subname/comments/123456/post_title_slug/234567/',
+    'https://www.reddit.com/r/subname/comments/123456/post_title_slug/234567',
+    'https://www.reddit.com/r/subname/comments/123456/post_title_slug/234567/',
+    'https://reddit.com/r/subname/comments/123456/post_title_slug/234567/?context=1337',
+    'https://www.reddit.com/r/subname/comments/123456/post_title_slug/234567/?context=1337',
+)
+
+
+NON_MATCHING_URLS = (
+    # we don't allow for query parameters on subreddit links (yet?)
+    'https://reddit.com/r/subname?param=value',
+    'https://reddit.com/r/subname/?param=value',
+    'https://www.reddit.com/r/subname?param=value',
+    'https://www.reddit.com/r/subname/?param=value',
+)
+
+
+@pytest.mark.parametrize('link', MATCHING_URLS)
+def test_url_matching(link, bot):
+    line = PreTrigger(bot.nick, ':User!user@irc.libera.chat PRIVMSG #channel {}'.format(link))
+    matches = bot.rules.get_triggered_rules(bot, line)
+
+    assert len([match for match in matches if match[0].get_plugin_name() == 'reddit']) == 1
```

Comment:
I think you should test *which* reddit rule is actually called, like `image_info` or `video_info`, etc.

Reply:
That was the original goal, but I simply couldn't find a way to get that information back from the list of triggered rules.

Reply:
Too busy at the time to really pay attention to details like that.

I think you can add a parameter to the `pytest.mark.parametrize` so that you get both the line and the expected name for the rule. You get the rule as the first parameter of match, and you can ask it its label through `get_rule_label`: that will return either the command name for a named rule, or the handler name for a generic rule (or the named label if `plugin.label` was used).

Reply:
Haven't a clue why I thought that wouldn't work. Maybe I was hung up on checking precisely which regex matched (handlers like `post_or_comment_info` have multiple matching patterns defined). No matter… Amended tests pushed.